File: all-submissions/2310-minimum-operations-to-halve-array-sum/solution.java
class Solution {
    public int halveArray(int[] nums) {

        int count = 0;
        double sum = 0;
        double halfSum = 0;
        double currSum = 0;
        PriorityQueue<Double> heap = new PriorityQueue<>(Comparator.reverseOrder());
        for(int num: nums) {
            heap.add((double)num);
            sum += (double)num;
        }
        currSum = sum;
        halfSum = sum/2;
        while(currSum > halfSum) {
            double halfVal = heap.remove()/2;
            currSum -= halfVal;
            heap.add(halfVal);
            count++;
        }
        return count;
        
    }
}

File: all-submissions/2310-minimum-operations-to-halve-array-sum/solution.py
class Solution:
    def halveArray(self, nums: List[int]) -> int:
        total = sum(nums)
        half = total/2
        heapq.heapify_max(nums)
        operations = 0
        print(total, half)
        while total > half:
            operations += 1
            max = heapq.heappop_max(nums)
            print(max)
            max_half = max/2
            total -= max_half
            heapq.heappush_max(nums, max_half)
        return operations
